cmd/main.go

```diff
@@ -105,6 +93,21 @@ func main() {
 		}
 	}
 
+	var secretType vault.SecretType
+
+	options := make(map[string]string, 0)
+
+	if *getCertificate && *commonName == "" {
+		log.Fatal("error: must supply common name when requesting certificate")
+	} else if *getCertificate {
+		secretType = vault.CertificateType
+
+		options["common_name"] = *commonName
+		options["ttl"] = *ttl
```

Comment:
could *ttl and *commonName == nil if not set? as they have no default value

Reply:
commonName and ttl are of type string so default to empty string

Added check to make sure commonName isn't an empty but according to vault api docs, ttl is allowed to be empty;

```ttl (string: "") - Specifies requested Time To Live. Cannot be greater than the role's max_ttl value. If not provided, the role's ttl value will be used. Note that the role values default to system values if not explicitly set.```